Cabal/Distribution/Version.hs

```diff
@@ -138,7 +150,24 @@ instance Text Version where
 mkVersion :: [Int] -> Version
 -- TODO: add validity check; disallow 'mkVersion []' (we have
 -- 'nullVersion' for that)
-mkVersion = Version
+mkVersion ns = case ns of
+    [] -> nullVersion
+    [v1]          | v1 <= 0xfffe
+      -> PV0 (mkW64 (v1+1) 0 0 0)
+    [v1,v2]       | v1 <= 0xfffe, v2 <= 0xfffe
+      -> PV0 (mkW64 (v1+1) (v2+1) 0 0)
+    [v1,v2,v3]    | v1 <= 0xfffe, v2 <= 0xfffe, v3 <= 0xfffe
+      -> PV0 (mkW64 (v1+1) (v2+1) (v3+1)  0)
+    [v1,v2,v3,v4] | v1 <= 0xfffe, v2 <= 0xfffe, v3 <= 0xfffe, v4 <= 0xfffe
```

Comment:
does it make any difference to do fewer comparisons, e.g. `v1 .|. v2 .|. v3 .|. v4 <= 0xfffe` (or is that trick correct since it's 0xfffe rather than 0xffff?)